### PR TITLE
[OPPRO-182] Change the Substrait plan for avg

### PR DESCRIPTION
**Pull request by rui-mo**
## What changes were proposed in this pull request?

This PR changed the Substrait plan for avg to avoid special handling in native. Changes are:

- Used sum and count to replace partial avg in Substrait plan.

- Added a row_constructor projection before aggregation for final avg.

- Seperated module for aggregation transformer.

Depends on: https://github.com/oap-project/velox/pull/31.

## How was this patch tested?

Verified on Jenkins.

### CODE REVIEW

**Comment by rui-mo** [comment]
@zhejiangxiaomai This PR depends on Velox [PR#31](https://github.com/oap-project/velox/pull/31), and is for the avg change we discussed. Please take a look if the upstream issue can be solved, thanks.

**Comment by zhejiangxiaomai** [comment]
> @zhejiangxiaomai This PR depends on Velox [PR#31](https://github.com/oap-project/velox/pull/31), and is for the avg change we discussed. Please take a look if the upstream issue can be solved, thanks.

Ok, I will test it.

File: backends-clickhouse/src/main/scala/io/glutenproject/execution/CHHashAggregateExecTransformer.scala [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+      } else {
+        var isPartial = true
+        aggregateExpressions.foreach(aggExpr => {
+          aggExpr.mode match {
+            case Partial =>
+            case _ => isPartial = false
+          }
+        })
+        isPartial
+      }
```
Comment (zzcclp) [inline backends-clickhouse/src/main/scala/io/glutenproject/execution/CHHashAggregateExecTransformer.scala:90]: Now ClickHouse Backend supports to use final agg, and use the function 'doValidate' of HashAggregateExecBaseTransformer directly.

Comment (rui-mo) [reply backends-clickhouse/src/main/scala/io/glutenproject/execution/CHHashAggregateExecTransformer.scala]: Fixed.

**Comment by rui-mo** [comment]
> LGTM, except a minor comment, remove the override function 'doValidate' of CHHashAggregateExecTransformer, and verified with ClickHouse Backend.

Thanks, removed the final agg flag.

**Pull request merged**